nntrainer/optimizers/adam.h

```diff
@@ -92,6 +91,8 @@ class Adam : public Optimizer {
   double getEpsilon() { return epsilon; }
 
 private:
+  static const std::string type;
```

Comment:
this type must belong to optimizer base class and extended class of adam.

Reply:
It is inferred through `getType()` not through object since virtual static member is not possible by nature.

Reply:
I meant from the purpose of baseclass being able to access it. I didn't realize base class (optimizer.h) has `getType()`.